File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        mp = {}
        for i, num in enumerate(nums):
            required = target - num
            if required in mp:
                return [mp[required], i]
            mp[num] = i
        return None

File: 1-two-sum/1-two-sum.cpp
#include <unordered_map>

using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> map;
        
        for (int i = 0; i < nums.size(); i++) {
            int required = target - nums[i];
            if (map.count(required) == 1)
                return vector<int>{map[required], i};
            map[nums[i]] = i;
        }
        return vector<int>();
    }
};